Cargo.toml

```diff
@@ -27,46 +27,37 @@ exclude = [
 ]
 
 [features]
-default = []
+default = ["compat-3-0-0", "rustls-tls", "dns-resolver"]
+compat-3-0-0 = []
 sync = []
-openssl-tls = ["openssl", "openssl-probe", "tokio-openssl"]
-
-# Enable support for v0.4 of the chrono crate in the public API of the BSON library.
-bson-chrono-0_4 = ["bson/chrono-0_4"]
-
-# Enable support for the serde_with crate in the BSON library.
-bson-serde_with = ["bson/serde_with"]
-
-# Enable support for v0.8 of the uuid crate in the public API of the BSON library.
-bson-uuid-0_8 = ["bson/uuid-0_8"]
-
-# Enable support for v1.x of the uuid crate in the public API of the BSON library.
-bson-uuid-1 = ["bson/uuid-1"]
+rustls-tls = ["dep:rustls", "dep:rustls-pemfile", "dep:tokio-rustls"]
+openssl-tls = ["dep:openssl", "dep:openssl-probe", "dep:tokio-openssl"]
+dns-resolver = ["dep:trust-dns-resolver", "dep:trust-dns-proto"]
```

Comment:
I'm not sure whether giving this feature a generic name is the right way (treating the actual dependency as an internal detail) or if it should just explicitly be called "trust-dns-resolver" in case we want to add other resolver crates in the future.

Reply:
Hmm, I think I'm in favor of keeping it generic. I was curious what the status of `trust-dns-resolver` was and it looks like they've actually "rebranded" to a different crate (see [here](https://docs.rs/trust-dns-resolver/latest/trust_dns_resolver/)) that we might want to switch over to at some point. Keeping the name neutral will make it easier for us to swap out the internal dep.

Worst case, if we did want to introduce more resolver crates we could treat this flag as our "default" choice and then allow users to opt into a different one if they want to.